request-manager/graph/grapher.go

```diff
@@ -217,25 +236,27 @@ func buildSeqGraph(seqSpec *spec.Sequence, idgen id.Generator) (seqGraph *Graph,
 		}
 	}
 
-	// Nodes we've yet to add
+	// Nodes we've yet to add, represented as single-node graphs
 	nodesToAdd := map[string]*Graph{}
 	for k, v := range nodes {
 		nodesToAdd[k] = v
 	}
 	nodesAdded := map[string]bool{}
 
 	// Build graph by adding nodes, starting from the source node, and then
-	// adding all adjacent nodes to the source node, and so on.
-	// We cannot add nodes in any order because we do not know the reverse dependencies
+	// adding all adjacent nodes to the source node, and so on
+	// We cannot add nodes in any order because we do not know the reverse
+	// dependencies
 	for len(nodesToAdd) > 0 {
 
-		componentAdded := false
+		nodeAdded := false
```

Comment:
Why do we presume this to be false from the start? (I know why, but future eng. might wonder.)